pkg/kubectl/cmd/log.go

```diff
@@ -137,28 +150,54 @@ func RunLog(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 		}
 	}
 
-	follow := false
-	if cmdutil.GetFlagBool(cmd, "follow") {
-		follow = true
+	logOptions := &api.PodLogOptions{
+		Follow:     cmdutil.GetFlagBool(cmd, "follow"),
+		Previous:   cmdutil.GetFlagBool(cmd, "previous"),
+		Timestamps: cmdutil.GetFlagBool(cmd, "timestamps"),
 	}
-
-	previous := false
-	if cmdutil.GetFlagBool(cmd, "previous") {
-		previous = true
+	if sinceDuration > 0 {
+		logOptions.SinceDuration = sinceDuration.String()
+	}
+	if t, err := api.ParseRFC3339(sinceTimestamp, util.Now); err == nil {
+		logOptions.SinceTimestamp = &t
+		logOptions.SinceDuration = ""
```

Comment:
You actually don't need this since if both sinceDuration and sinceTimestamp have been specified you will return a usage error above.